vm/src/vm/vm_ops.rs

```diff
@@ -174,160 +177,348 @@ impl VirtualMachine {
         })
     }
 
-    pub fn _sub(&self, a: &PyObject, b: &PyObject) -> PyResult {
-        self.call_or_reflection(
-            a,
-            b,
-            identifier!(self, __sub__),
-            identifier!(self, __rsub__),
-            |vm, a, b| Err(vm.new_unsupported_binop_error(a, b, "-")),
-        )
+    fn binary_op1<F>(
+        &self,
+        a: &PyObject,
+        b: &PyObject,
+        op_slot: PyNumberMethodsOffset,
+        unsupported: F,
+    ) -> PyResult
+    where
+        F: Fn(&VirtualMachine, &PyObject, &PyObject) -> PyResult,
+    {
+        let num_a = a.to_number();
+        let num_b = b.to_number();
+
+        let slot_a = num_a.methods(&op_slot, self)?.load();
+        let slot_b = num_b.methods(&op_slot, self)?.load();
+
+        if let Some(slot_a) = slot_a {
+            if let Some(slot_b) = slot_b {
+                // Check if `a` is subclass of `b`
```

Comment:
@youknowone 

According to CPython implementation https://github.com/python/cpython/blob/main/Objects/abstract.c#L842 , we may need to ensure `b` is a "strict subclass" of `a` in order to use `slot_b` for operation.

`fast_isinstance` method seems including the situation that `a` & `b` belong to the same class, https://github.com/RustPython/RustPython/blob/main/vm/src/object/ext.rs#L459 .
Is there a method or proper approach to check "strict subclass" please?

Reply:
I don't think we have a function to do that.
But how about `!a.is(b) && a.fast_issubclass(b)`?
when a and b are identical type, `a.is(b)` will be true